### PR TITLE
[WFLY-13490] Add Galleon layer for web console

### PR DESCRIPTION
**Pull request by yersan**
This supersedes: https://github.com/wildfly/wildfly/pull/13302

It is the same PR but this one includes the integration tests available on WildFly tests suite for the Web Console.

One of those tests is ignored in the case of layers is enabled. The ignored test verifies header overriding for a non management endpoint, using the metrics endpoint. However when the web console is supplied via a Galleon Layer there is no other non management endpoint, so the test is ignored for layers provisioning.

Jira issue: https://issues.redhat.com/browse/WFLY-13490
Proposal: https://github.com/wildfly/wildfly-proposals/pull/312

### CODE REVIEW

**Comment by yersan** [comment]
Added also `org.jboss.as.patching` since it is necessary to supply the `/core-service=patching` generated at runtime.

**Comment by yersan** [comment]
@bstansberry @hpehl would you mind taking a look and approve in case? I don't plan to apply code changes as it is now. Feature issue is in "Review" state, thanks!

File: ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="web-console">
+    <dependencies>
+        <layer name="secure-management"/>
+        <layer name="core-management"/>
+    </dependencies>
+    <packages>
+        <package name="org.jboss.as.console"/>
+        <package name="org.jboss.as.patching"/>
```
Comment (bstansberry) [inline ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml:9]: @yersan Why this?

File: ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="web-console">
+    <dependencies>
+        <layer name="secure-management"/>
+        <layer name="core-management"/>
```
Comment (bstansberry) [inline ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml:5]: Why this?

File: ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml (hunk 1 above)
Comment (yersan) [reply ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml:9]: HAL is not prepared to show the "Pathching" menu entry if `/core-service=patching` resource does not exist.

It is a harmless error, we can live with it, the user would see an `"WFLYCTL0030: No resource definition is registered for address [(\"core-service\" => \"patching\")]"`, which indeed is fine. I don't know if we plan to use patching in the future for a bootable jar installation.

I just added it to avoid displaying this error, however, probably the users will receive a different one if they try to patch a bootable jar. The `org.jboss.as.patching` package has no impact on the disk footprint.

What do you prefer to do with this package?
Now I'm more inclined to remove it and live with the error.

File: ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml (hunk 2 above)
Comment (bstansberry) [reply ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml:5]: I'm fine with keeping this but a HAL issue to allow it to gracefully live without this would be good.

File: ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml (hunk 1 above)
Comment (bstansberry) [reply ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml:9]: @yersan IIRC with an RPM installation EAP doesn't install this (because RPM-based EAP is updated using RPM), so HAL should be able to handle its absence gracefully. I'd prefer that HAL corrects this.

Comment (yersan) [reply ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml:9]: https://issues.redhat.com/browse/HAL-1706

File: ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml (hunk 2 above)
Comment (yersan) [reply ee-galleon-pack/src/main/resources/layers/standalone/web-console/layer-spec.xml:5]: https://issues.redhat.com/browse/HAL-1707

**Comment by yersan** [comment]
retest this please

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc [hunk 3]
```diff
@@ -165,6 +165,10 @@ Layers that you combine with "use-case tailored" layers to extend the capabiliti
 * _observability_: Support for MicroProfile monitoring and configuration features. 
 Includes Health support (optional), _microprofile-config_ (optional), _microprofile-metrics_ (optional) and _open-tracing_ (optional).
 
+Tools layers:
+
+* _web-console_: Support for HAL web console.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc:170]: @hpehl WDYT about wording more like this:

"Support for loading the HAL web console from the /console context on the HTTP management interface. Not required to use a HAL console obtained independently and configured to connect to the server."

Maybe too wordy. Or not wordy enough. :) I'm just wanting to get across that this isn't needed for HAL to work; it just gives you a copy of HAL.

File: ee-feature-pack/galleon-common/src/main/resources/layers/standalone/web-console/layer-spec.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="web-console">
+    <dependencies>
+        <layer name="secure-management"/>
```
Comment (yersan) [inline ee-feature-pack/galleon-common/src/main/resources/layers/standalone/web-console/layer-spec.xml:4]: Replace this by `management` once https://github.com/wildfly/wildfly-core/pull/4296 is merged

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc (hunk 3 above)
Comment (hpehl) [reply docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc:170]: @bstansberry 
Sounds good to me. I don't think it's too wordy.

**Comment by tommaso-borgato** [comment]
@bstansberry tagging you in case you want to give a look

File: ee-feature-pack/galleon-common/src/main/resources/layers/standalone/web-console/layer-spec.xml (hunk 4 above)
Comment (bstansberry) [reply ee-feature-pack/galleon-common/src/main/resources/layers/standalone/web-console/layer-spec.xml:4]: Actually this can be done once https://github.com/wildfly/wildfly-core/pull/4295 is merged and available, and it is.

**Comment by bstansberry** [comment]
Superseded by #13520, which just adjusts the two items discussed in comments here.
